datadog_checks_base/datadog_checks/base/checks/base.py

```diff
@@ -308,8 +311,23 @@ def __init__(self, *args, **kwargs):
 
         self.__formatted_tags = None
         self.__logs_enabled = None
+        self.__fips_mode = 0
 
+        # On Windows, FIPS mode is activated through a registry
+        # https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp4825.pdf
+        if sys.platform == "win32":
+            try:
+                import winreg
+                with winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, "HKLM\System\CurrentControlSet\Control\Lsa\FIPSAlgorithmPolicy\STE") as key:
+                    fips_registry, _ = winreg.QueryValueEx(key, FIPS_REGISTRY_KEY)
```

Comment:
interesting, I see the key you had is referenced in the doc you linked but I've never seen it before, and it doesn't seem to have any effect on my Windows Server 2022 VM.

I can't find any other Microsoft references to it besides this Q/A thread that's also confused about it
https://learn.microsoft.com/en-us/answers/questions/1183237/use-fips-compliant-algorithms-windows-11-22h2-slow

For now lets stick to the one documented by Go and that seems to have an effect on Windows
https://github.com/microsoft/go/blob/microsoft/main/eng/doc/fips/README.md#windows-fips-mode-cng

```suggestion
                with winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, "HKLM\System\CurrentControlSet\Control\Lsa\FIPSAlgorithmPolicy") as key:
                    fips_registry, _ = winreg.QueryValueEx(key, "Enabled")
```